pkg/networkservice/utils/failure/server.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2020-2021 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package failure provides chain elements failing request at specified times
+package failure
```

Comment:
Probably we need to add some kind of alert in style 'DO NOT USE THIS IN PRODUCTION ;)

Can we just use `injecterror` instead of `failure`?

Reply:
`failure.NewServer(-1)` is the same as `injecterror.NewServer()`. So we can think about replacing `injecterror` with `failure`. Do we need it?

Reply:
Why we cannot replace failure with `injecterror`?

Reply:
Because `injecterror` returns an error every request. `failure` return an error only on `failureTimes` requests.
`injecterror.NewServer()` - error, error, error, ...
`failure.NewServer(1, 3, -1)` - success, error, success, error, error, ...

Reply:
Probably we could add options for `injecterror` to avoid creating a similar entity.

Reply:
Yep, reworked.